hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/GeneratedValuesProcessor.java

```diff
@@ -49,24 +56,23 @@ public class GeneratedValuesProcessor {
 	private final List<AttributeMapping> generatedValuesToSelect;
 	private final JdbcParametersList jdbcParameters;
 
-	private final EntityMappingType entityDescriptor;
-	private final SessionFactoryImplementor sessionFactory;
+	private final EntityPersister entityDescriptor;
 
 	public GeneratedValuesProcessor(
-			EntityMappingType entityDescriptor,
```

Comment:
Please re-introduce this constructor and mark it as deprecated.

Reply:
This is internal, should we care about constructors?